src/Resources/Interfaces/ResourceProxyInterface.php

```diff
@@ -10,9 +10,6 @@ interface ResourceProxyInterface
 {
     /**
      * Set the API with which to fetch the proxied resource.
-     *
-     * @param MyParcelComApiInterface $api
-     * @return $this
      */
-    public function setMyParcelComApi(MyParcelComApiInterface $api);
+    public function setMyParcelComApi(?MyParcelComApiInterface $api): self;
```

Comment:
Tests were failing because the `ResourceFactory` was calling this method for resources that "Proxy" their relationships. But you're right, we should fix the setup of those tests instead of making this function accept null.